tensorflow/contrib/distribute/python/parameter_server_strategy.py

```diff
@@ -294,9 +294,12 @@ def _call_for_each_tower(self, fn, *args, **kwargs):
   def _verify_destinations_not_different_worker(self, destinations):
     if destinations is None:
       return
+    task_id = self._task_id
+    if task_id is None:
+      task_id = 0
     for d in cross_tower_ops_lib.get_devices_from(destinations):
       d_spec = tf_device.DeviceSpec.from_string(d)
-      if d_spec.job == self._task_type and d_spec.task != self._task_id:
+      if d_spec.job == self._task_type and d_spec.task != task_id:
```

Comment:
Could you add a unit test for this?

Reply:
@rmlarsen Added unit test. Please kindly review